edx_when/api.py

```diff
@@ -175,7 +175,7 @@ def get_dates_for_course(course_id, user=None, use_cached=True, schedule=None):
             need_schedule = False
             schedule = cdate.schedule_for_user(user)
 
-        key = (cdate.location, cdate.field)
+        key = (cdate.location.replace(course_key=course_id), cdate.field)
```

Comment:
Well, okay, nit, we do have: `map_into_course` on keys for this

Reply:
Ah, I thought there was a method for that, but couldn't find it.